Get pool data however the state (resp. status)

samy-mahmoudi opened the pull request:
This commit substitutes 'zpool iostat' for 'zpool status' to fix improper parsing due to the variety of states and statuses.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppetlabs-zfs_core?pullRequest=55) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-zfs_core?pullRequest=55) before we can accept your contribution.<br/><hr/>**Samy Mahmoudi** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-zfs_core?pullRequest=55) it.</sub>

samy-mahmoudi committed:
Get pool data however the state (resp. status)

This commits substitutes 'zpool iostat' for 'zpool status' to
fix improper parsing due to the variety of states and statuses.

johro81 commented on lib/puppet/provider/zpool/zpool.rb:
```diff
@@ -62,19 +62,21 @@ def process_zpool_data(pool_array)
-    zpool_data = out.lines.select { |line| line.index("\t") == 0 }.map { |l| l.strip.split("\s")[0] }
-    zpool_data.shift
+    out = execute("zpool iostat -v #{zpool_opts} #{@resource[:pool]}", failonfail: false, combine: false)
+    zpool_data = out.lines[3..-2].map { |line| line.strip.split("\s")[0] }
```
This fails on Solaris 11 since the last separator is included because of an empty line at the end of the command.

johro81 reviewed (approved):
I can confirm that these changes work for  my case with Solaris 11.

Update: does not work if zpool is missing:
```
# zpool iostat -d test
cannot open 'test': no such pool
```
this leads to undefined method `map' for nil:NilClass
